services/movie_session.py

```
@@ -41,4 +45,13 @@ def update_movie_session(
 
 
 def delete_movie_session_by_id(session_id: int) -> None:
-    MovieSession.objects.get(id=session_id).delete()
+    if session_id:
+        MovieSession.objects.get(id=session_id).delete()
+
+    if not session_id:
+        raise f"The id: {session_id} Does Not Exist"
```

The `delete_movie_session_by_id` function has a logical flaw. The check `if session_id:` is unnecessary since `session_id` is a required parameter. Additionally, the exception message in line 52 is incorrect; it should raise a proper exception like `DoesNotExist` or `ValueError` instead of a string.